front/src/applications/operationalStudies/utils.ts

```diff
@@ -235,12 +235,17 @@ export const isScheduledPointsNotHonored = (
   }
 
   if (!trainSchedule.schedule) return false;
-  return trainSchedule.schedule.some((schedule, index) => {
+  return trainSchedule.schedule.some((schedule) => {
     if (!schedule.arrival) return false;
-    // need to add +1 because origin will never be in schedules
+    const matchindIndex = trainSchedule.path.findIndex((pathItem) => pathItem.id === schedule.at);
```

Comment:
This solution gives quadratic complexity. I don't know if it's a problem, but when we set up this logic we thought about this complexity problem.
@flomonster, what do you think?

Reply:
Complexity can easily be improved. Just create a mapping from “pathItem.id” -> “pathItem”.

This way you can access the path item with its ID and check whether it was found.